### PR TITLE
Fixed loading ZoneCylinder texture

### PR DESCRIPTION
**Pull request by NFSMONSTR**
It seems like that texture for ZoneCylinder loaded correctly but `texturelookup ` can't find texture because wrong texture slot is set. So in `loadClump` we should set it manually

### CODE REVIEW

File: rwengine/src/engine/GameData.cpp [hunk 1]
```diff
@@ -409,6 +409,15 @@ ClumpPtr GameData::loadClump(const std::string& name) {
     return m;
 }
 
+ClumpPtr GameData::loadClump(const std::string& name, const std::string& textureSlot) {
+    std::string currentSlot = currenttextureslot;
+    if (textureSlot.size()>0)
```
Comment (ShFil119) [inline rwengine/src/engine/GameData.cpp]: Maybe better `!textureSlot.empty()`?

File: rwgame/RWGame.cpp [hunk 2]
```diff
@@ -26,10 +26,10 @@
 #include <iomanip>
 #include <iostream>
 
-std::map<GameRenderer::SpecialModel, std::string> kSpecialModels = {
-    {GameRenderer::ZoneCylinderA, "zonecyla.dff"},
-    {GameRenderer::ZoneCylinderB, "zonecylb.dff"},
-    {GameRenderer::Arrow, "arrow.dff"}};
+std::map<GameRenderer::SpecialModel, std::pair<std::string,std::string>> kSpecialModels = {
```
Comment (ShFil119) [inline rwgame/RWGame.cpp]: Can it be set as const?

Comment (NFSMONSTR) [reply rwgame/RWGame.cpp]: You mean the whole `std::map`?

Comment (ShFil119) [reply rwgame/RWGame.cpp]: yeah

Comment (NFSMONSTR) [reply rwgame/RWGame.cpp]: I'll try. Thank you

File: rwengine/src/engine/GameData.cpp [hunk 3]
```diff
@@ -409,6 +409,15 @@ ClumpPtr GameData::loadClump(const std::string& name) {
     return m;
 }
 
+ClumpPtr GameData::loadClump(const std::string& name, const std::string& textureSlot) {
+    std::string currentSlot = currenttextureslot;
+    if (!textureSlot.empty())
```
Comment (ShFil119) [inline rwengine/src/engine/GameData.cpp]: Btw, does this `if` change anything? I mean you save current state of `currenttextureslot` in `currentSlot`. After that you override it or not. But at the end you use value from `currentSlot`. (or I don't notice something?)

Comment (NFSMONSTR) [reply rwengine/src/engine/GameData.cpp]: `currenttextureslot` is global variable. It used inside of loadClump to determine which texture slot to use to search for texture for this model




```c++
(GameData.cpp)
...
GameData::GameData(Logger* log, const rwfs::path& path)
    : datpath(path), logger(log), engine(nullptr) {
    dffLoader.setTextureLookupCallback(
        [&](const std::string& texture, const std::string&) {
            return findSlotTexture(currenttextureslot, texture);
        });
}
```

Comment (ShFil119) [reply rwengine/src/engine/GameData.cpp]: Thx for explanation.

**Pull request merged**